corehq/apps/ota/views.py

```diff
@@ -169,7 +169,20 @@ def phone_holds_all_cases(request):
         except MissingSyncLog:
             return False
 
-    if not case_ids_to_claim and phone_holds_all_cases(request):
+    def cases_on_phone_modified_since_last_synclog_date(request):
```

Comment:
This should only need to check the cases that are being claimed, not every case on the phone.

You can also use `CommCareCase.objects.get_modified_case_ids(domain, case_ids, sync_log)`

Reply:
ah makes sense [dc0a6d4](https://github.com/dimagi/commcare-hq/pull/32554/commits/dc0a6d4809626a8a0d7025755bee4a7927072aa4)